pac4j-cas/src/main/java/org/pac4j/cas/client/CasClient.java

```diff
@@ -1,388 +1,204 @@
-        final Cas30ProxyTicketValidator cas30ProxyTicketValidator = (Cas30ProxyTicketValidator) this.ticketValidator;
-        cas30ProxyTicketValidator.setEncoding(this.encoding);
-        cas30ProxyTicketValidator.setAcceptAnyProxy(this.acceptAnyProxy);
-        cas30ProxyTicketValidator.setAllowedProxyChains(this.allowedProxyChains);
-        if (this.casProxyReceptor != null) {
-            cas30ProxyTicketValidator.setProxyCallbackUrl(this.casProxyReceptor.computeFinalCallbackUrl(context));
-            cas30ProxyTicketValidator.setProxyGrantingTicketStorage(this.casProxyReceptor
-                    .getProxyGrantingTicketStorage());
-        }
-    }
-
-    protected void initializeCas30Protocol(final WebContext context) {
-        this.ticketValidator = new Cas30ServiceTicketValidator(this.casPrefixUrl);
-        final Cas30ServiceTicketValidator cas30ServiceTicketValidator = (Cas30ServiceTicketValidator) this.ticketValidator;
-        cas30ServiceTicketValidator.setEncoding(this.encoding);
-        if (this.casProxyReceptor != null) {
-            cas30ServiceTicketValidator.setProxyCallbackUrl(this.casProxyReceptor.computeFinalCallbackUrl(context));
-            cas30ServiceTicketValidator.setProxyGrantingTicketStorage(this.casProxyReceptor
-                    .getProxyGrantingTicketStorage());
-        }
-    }
-
-    protected void initializeCas20ProxyProtocol(final WebContext context) {
-        this.ticketValidator = new Cas20ProxyTicketValidator(this.casPrefixUrl);
-        final Cas20ProxyTicketValidator cas20ProxyTicketValidator = (Cas20ProxyTicketValidator) this.ticketValidator;
-        cas20ProxyTicketValidator.setEncoding(this.encoding);
-        cas20ProxyTicketValidator.setAcceptAnyProxy(this.acceptAnyProxy);
-        cas20ProxyTicketValidator.setAllowedProxyChains(this.allowedProxyChains);
-        if (this.casProxyReceptor != null) {
-            cas20ProxyTicketValidator.setProxyCallbackUrl(this.casProxyReceptor.computeFinalCallbackUrl(context));
-            cas20ProxyTicketValidator.setProxyGrantingTicketStorage(this.casProxyReceptor
-                    .getProxyGrantingTicketStorage());
-        }
-    }
 
-    protected void initializeCas20Protocol(final WebContext context) {
-        this.ticketValidator = new Cas20ServiceTicketValidator(this.casPrefixUrl);
-        final Cas20ServiceTicketValidator cas20ServiceTicketValidator = (Cas20ServiceTicketValidator) this.ticketValidator;
-        cas20ServiceTicketValidator.setEncoding(this.encoding);
-        if (this.casProxyReceptor != null) {
-            cas20ServiceTicketValidator.setProxyCallbackUrl(this.casProxyReceptor.computeFinalCallbackUrl(context));
-            cas20ServiceTicketValidator.setProxyGrantingTicketStorage(this.casProxyReceptor
-                    .getProxyGrantingTicketStorage());
-        }
-    }
+        configuration.setCallbackUrlResolver(this.getCallbackUrlResolver());
```

Comment:
@mmoayyed @millross the `CasClient` has been broken into small subcomponents and it's really a good idea: makes things clearer and reusable (Client -> ClientV2).